src/main/java/org/dave/compactmachines3/tile/TileEntityFieldProjector.java

```diff
@@ -280,105 +269,83 @@ public boolean shouldRenderField() {
+        if (!this.isMaster()) // Only run the rest of the recipe code on the master
             return;
-        }
 
         BlockPos center = this.getPos().offset(this.getDirection(), magnitude * 2);
 
-        // Determine whether there is a catalyst item inside this projectors field
-        double growWD = magnitude + 0.5d;
-        BlockPos centerPosOfField = this.getPos().offset(this.getDirection(), magnitude);
-        AxisAlignedBB centerBB = new AxisAlignedBB(centerPosOfField).grow(0, growWD, 0);
-        if(getDirection() == EnumFacing.NORTH || getDirection() == EnumFacing.SOUTH) {
-            centerBB = centerBB.grow(growWD, 0, 0);
-        } else {
-            centerBB = centerBB.grow(0, 0, growWD);
-        }
+        // Determine whether there is a catalyst item inside the entire field
+        double growWD = magnitude + 1.5d;
+        AxisAlignedBB fieldBB = new AxisAlignedBB(center).grow(growWD);
+        List<EntityItem> items = world.getEntitiesWithinAABB(EntityItem.class, fieldBB);
 
-        List<EntityItem> items = world.getEntitiesWithinAABB(EntityItem.class, centerBB);
-
-        if(isMaster()) {
-            BlockPos centerOfTopField = this.getPos().offset(this.getDirection(), magnitude*2).offset(EnumFacing.UP, magnitude);
-            AxisAlignedBB topBB = new AxisAlignedBB(centerOfTopField).grow(magnitude + 0.5d, 0, magnitude + 0.5d);
-            items.addAll(world.getEntitiesWithinAABB(EntityItem.class, topBB));
-        }
-
-        for(EntityItem item : items) {
-            if(item.ticksExisted > ConfigurationHandler.Settings.maximumCraftingCatalystAge) {
+        for (EntityItem item : items) {
+            if (item.ticksExisted > ConfigurationHandler.Settings.maximumCraftingCatalystAge) {
                 continue;
             }
 
             MultiblockRecipe multiblockRecipe = MultiblockRecipes.tryCrafting(world, getPos(), item.getItem());
-            if(multiblockRecipe == null) {
+            if (multiblockRecipe == null) {
                 continue;
             }
 
             // Remove blocks from the world
-            for(BlockPos pos : getInsideBlocks()) {
+            for (BlockPos pos : getInsideBlocks()) {
                 world.setBlockToAir(pos);
-                ((WorldServer)world).spawnParticle(EnumParticleTypes.SMOKE_LARGE, pos.getX(), pos.getY(), pos.getZ(), 10, 0.5D, 0.5D, 0.5D, 0.01D);
+                ((WorldServer) world).spawnParticle(EnumParticleTypes.SMOKE_LARGE, pos.getX(), pos.getY(), pos.getZ(), 10, 0.5D, 0.5D, 0.5D, 0.01D);
             }
 
             // Reduce the number of items in the stack
             ItemStack stack = item.getItem();
-            if(stack.getCount() == 1) {
-                item.setDead();
-            } else if(stack.getCount() > 1) {
+            if (stack.getCount() > 1) {
                 stack.setCount(stack.getCount() - 1);
                 item.setItem(stack.copy());
+            } else {
+                item.setDead();
```

Comment:
Change the comment here, `setDead` is unclear but good comments are still useful